app/services/workarea/update_variant_components.rb

```diff
@@ -0,0 +1,57 @@
+module Workarea
+  class UpdateVariantComponents
+    def initialize(variant, components_params)
+      @variant = variant
+      @components_params = components_params.presence || []
+    end
+
+    def perform
+      @components_params.each do |params|
+        params.deep_symbolize_keys! if params.respond_to?(:deep_symbolize_keys!)
+
+        if remove?(params)
```

Comment:
Passing params so many times makes me feel like there could be a class to clean this up

```
class ComponentParams
  def apply
    remove if remove?
    update if update?
  end
end
```

Reply:
fair. I'll explore